std/datetime.d

```diff
@@ -29161,6 +29191,72 @@ private:
         _hasDST = hasDST;
     }
 
+    // Android concatenates the usual timezone directories into a single file,
+    // tzdata, along with an index to jump to each timezone's offset.  In older
+    // versions of Android, the index was stored in a separate file, zoneinfo.idx,
+    // whereas now it's stored at the beginning of tzdata.
+    version(Android)
+    {
+        // Keep track of whether there's a separate index, zoneinfo.idx
+        static bool separate_index;
```

Comment:
Hmm, I hadn't though about multi-threaded use of this.

What's there now will be thread-safe, as static is TLS.  I suppose the only issue you're raising is efficiency, where this and the following static variables will be set separately in each thread that initializes timezone data, implying some unnecessary repetition.  The index is parsed so fast that I don't think it will matter generally, but I'm open to trying to make it more efficient if wanted, by making everything immutable and only taking the hit once, albeit imposed whether one uses timezone data or not.